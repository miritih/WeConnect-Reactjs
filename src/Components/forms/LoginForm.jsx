import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import InputField from '../inputs/InputField';
import Button from '../inputs/Button';

const LoginForm = ({
	handleChange, handleSubmit, username, password, loading,
}) => {
	return (
		<div className="container wrapper">
			<div className="main-center">
				<div className="card">
					<div className="card-body">
						<img className="profile-img" src="//ssl.gstatic.com/accounts/ui/avatar_2x.png" alt="login" />
						<form className="form" onSubmit={handleSubmit}>
							<InputField
								type="text"
								name="username"
								label="Username"
								onChange={handleChange}
								placeholder="enter username"
								value={username}
								required="required"
							/>
							<InputField
								type="password"
								name="password"
								label="Password"
								required="required"
								onChange={handleChange}
								placeholder="enter password"
								value={password}
							/>
							<Button
								type="submit"
								className="btn btn-lg btn-success btn-block btn-signin"
								loading={loading}
								text={loading ? 'loading... ' : 'Login '}
								disabled={loading ? 'disabled' : null}
							/>
							<p className="message">Not registered? <Link to="/register">Create an account. </Link>
								<span> Forgot password?  <Link to="/forgotpass">Reset</Link> </span>
							</p>
						</form>
					</div>
				</div>
			</div>
		</div>
	);
};
LoginForm.propTypes = {
	handleChange: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	username: PropTypes.string,
	password: PropTypes.string,
};
LoginForm.defaultProps = {
	username: '',
	password: '',
};

export default LoginForm;
