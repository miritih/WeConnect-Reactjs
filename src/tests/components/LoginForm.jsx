import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LoginForm from '../../Components/forms/Login';
// define glabal imports
/* global it expect, jest describe */

describe('LoginForm', () => {
  let props;
  let wrapper;
  // setup method

  // test my whole form dom elements using snapshot
  it('renders without crashing', () => {
    const pathname = {
      pathname: '/',
    };
    const rendered = renderer.create(<Router><LoginForm location={pathname} /></Router>);

    expect(rendered.toJSON()).toMatchSnapshot();
  });

  // Test that LoginForm component has a form
  it('should have a `<form>` element', () => {
    wrapper = shallow(<LoginForm {...props} />);
    expect(wrapper.find('form').length).toBe(1);
  });

  // =================
  // Tests form logic
  // ================

  it('should call handlesubmit when form is submitted', () => {
    const handleSubmit = jest.fn();
    wrapper = mount(<Router><LoginForm handleSubmit={handleSubmit} /></Router>);
    const form = wrapper.find('form');
    form.simulate('submit', handleSubmit);
    expect(handleSubmit).toBeCalled();
  });

  it('Input should call handleChange it changes', () => {
    const handleChange = jest.fn();
    const event = {
      target: { value: 'the-value' },
    };
    wrapper = shallow(<LoginForm />);
    const input = wrapper.find('form').childAt(1);
    console.log(input.props());
    input.simulate('change', event);
    expect(handleChange).toBeCalled();
  });
});
