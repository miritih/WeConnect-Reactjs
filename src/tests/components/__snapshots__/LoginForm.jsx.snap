// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders without crashing 1`] = `
<div
  className="container wrapper"
>
  <div
    className="main-center"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <img
          alt="login"
          className="profile-img"
          src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
        />
        <p
          className="profile-name-card"
          id="profile-name"
        />
        <form
          className="form"
          onSubmit={[Function]}
        >
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="form-control"
            name="username"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            htmlFor="password"
          >
            password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            type="text"
            value=""
          />
          <br />
          <button
            className="btn btn-lg btn-success btn-block btn-signin"
            type="submit"
          >
            Login
          </button>
          <p
            className="message"
          >
            Not registered? 
            <a
              href="/register"
              onClick={[Function]}
            >
              Create an account
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
